ops/charm.py

```diff
@@ -34,30 +34,31 @@
 from ops.framework import EventBase, EventSource, Framework, Object, ObjectEvents
 
 if TYPE_CHECKING:
-    from typing_extensions import Literal, Required, TypedDict
+    from typing import Literal
+
+    from typing_extensions import Required, TypedDict
 
     from ops.framework import Handle, JsonObject, _SerializedData
     from ops.model import Container, Numerical, Relation, Storage
 
     # CharmMeta also needs these.
     _ActionParam = Dict[str, 'JsonObject']  # <JSON Schema definition>
-    _ActionMetaDict = TypedDict(
-        '_ActionMetaDict', {
-            'title': str,
-            'description': str,
-            'params': Dict[str, _ActionParam],
-            'required': List[str]},
-        total=False)
+
+    class _ActionMetaDict(TypedDict, total=False):
```

Comment:
This is the first that I've seen meta arguments to classes (`total=False`).
Apparently __init_subclass__ has been a thing since about python 3.6.

Reply:
From the arg, I'd suspect that `TypedDict` is `@dataclass`, though I haven't actually looked at the class definition.

Do we have use for pseudo-metaclass stuff other than the `__set_name__` changes from the last PR (part of the same PEP as `__init_subclass__`, IIRC)?

Reply:
Keep up with the times, @jameinel. ;-) Nah, seriously, Python moves way too fast these days IMO, particularly in the typing world.